v2/client/client.go

```diff
@@ -228,6 +228,10 @@ func (c *ceClient) StartReceiver(ctx context.Context, fn interface{}) error {
 					return
```

Comment:
That's weird, when the context is closed the receiver or the responder returns `io.EOF`. Which protocol are you using? To me it sounds like a problem of the protocol you're using which doesn't properly fullfill the contract.

Reply:
I'm using the pubsub protocol

Reply:
Then the pubsub protocol is bugged, because it should return io.EOF when the context is closed. Please change name and description of #580 so the pubsub protocol maintainers can take care of it

Reply:
Issue #580 updated and created a new PR #582